core/src/main/java/io/grpc/internal/DnsNameResolverProvider.java

```diff
@@ -49,12 +51,16 @@ public DnsNameResolver newNameResolver(URI targetUri, Attributes params) {
       Preconditions.checkArgument(targetPath.startsWith("/"),
           "the path component (%s) of the target (%s) must start with '/'", targetPath, targetUri);
       String name = targetPath.substring(1);
+      ProxyDetector proxyDetector = params.get(Factory.PARAMS_PROXY_DETECTOR);
+      if (proxyDetector == null) {
+        proxyDetector = GrpcUtil.NOOP_PROXY_DETECTOR;
```

Comment:
GrpcUtil.getDefaultProxyDetector  ?

Reply:
I had suggested the NOOP_PROXY_DETECTION here as a temporary thing (some logic was moved from the Builder). I figured @zhangkun83 (or whomever does #4159) would remove it when we replace the Attributes with a concrete class.

Reply:
No problem @ejona86 , applied @zhangkun83 suggestions